main/src/main/java/cgeo/geocaching/filters/gui/GeocacheFilterActivity.java

```diff
@@ -202,7 +202,9 @@ private void initializeStorageOptions() {
                                     }
                                 });
                     })
-                    .setChoiceMode(SimpleItemListModel.ChoiceMode.SINGLE_PLAIN);
+                    .setChoiceMode(SimpleItemListModel.ChoiceMode.SINGLE_PLAIN)
+                    .activateGrouping(f -> FilterUtils.getGroupFromFilterName(f.getName()))
+                    .setGroupDisplayMapper(gi -> TextParam.text("**" + gi.getGroup() + "** *(" + gi.getContainedItemCount() + ")*").setMarkdown(true));;
```

Comment:
You might want to allow grouping of groups by also setting .`setGroupGroupMapper`.

For a full example on what's possible see:
https://github.com/cgeo/cgeo/blob/9612d63941e9ae277620d254a5cc64d8e5f82d01/main/src/main/java/cgeo/geocaching/list/StoredList.java#L189-L202